codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/Instantiator.kt

```diff
@@ -370,3 +376,23 @@ open class Instantiator(
         else -> throw CodegenException("Unrecognized shape `$shape`")
     }
 }
+fun StructureShape.builderSymbol(symbolProvider: RustSymbolProvider): Symbol {
```

Comment:
We should add a `symbolForBuilder(shape: Shape)` to `RustSymbolProvider` instead of this approach (there are examples in there for operation error and event stream error symbols). The module location needs to be configurable for my crate reorganization work.